### PR TITLE
A11y updates for ORA UI Studio Messaging 

### PR DESCRIPTION
**Pull request by nsprenkle**
**TL;DR -** Collection of accessibility-related tweaks for the ORA editor views in Studio.

JIRA: [AU-91](https://openedx.atlassian.net/browse/AU-91) and [AU-92](https://openedx.atlassian.net/browse/AU-92)

**What changed?**

- Validation alert banner updates:
  - Add `aria-label=”warning”` to alert icon.
  - Add `role="alert"` to banner.
  - Add `aria-label=”close alert”` to close icon. Confirmed it is focusable and closable via keyboard.
- Tab updates:
  - Add `role="tab"`
  - Add screen reader text identifying the number of errors per tab.
  - Move warning icons to HTML elements (instead of pseudo-styling) and add `aria-label=””`
- Validation updates:
  - Add `aria-invalid="true"` to fields failing validation for each edit page.
- Updated some icon, text, and banner colors for better contrast (RE WCAG guidelines). See screenshots.
- Various refactors required to support the above.
- Unrelated doc updates. 

**Screenshots**
ora-validation-icon-refresh

**Developer Checklist**

- [x] Reviewed the [release process](https://github.com/edx/edx-ora2/blob/master/.github/release_process.md)
- [x] Translations and JS/SASS compiled
- [x] Bumped version number in [setup.py](https://github.com/edx/edx-ora2/blob/a62e81a9b0d89223476967ec3c27f3557a850735/setup.py#L39) and [package.json](https://github.com/edx/edx-ora2/blob/a62e81a9b0d89223476967ec3c27f3557a850735/package.json#L3)

**Testing Instructions**

1. In Studio > Course > ORA, click the "Edit" button to open the settings for an ORA.
2. For each tab (except "Prompt" which never fails validation):
  1. Introduce error(s) (see  https://github.com/edx/edx-ora2/pull/1662 for examples)
  2. Verify warning icons appear on tabs correctly
  3. In the browser inspection tools, verify:
     1. Screen-reader text shows number of errors.
     2. ARIA fields show invalid markup on fields failing validation
3. Verify other ARIA fields are correct (as listed in "What Changed?")

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [ ] I've done a visual code review
- [ ] I've tested the new functionality

FYI: @edx/masters-devs-gta

### CODE REVIEW

File: openassessment/templates/openassessmentblock/edit/oa_edit_header_and_validation.html [hunk 1]
```diff
@@ -2,22 +2,51 @@
...
     <h6 id="oa_editor_window_title" class="title modal_window_title" >{% trans "Open Response Assessment" %}</h6>
     <ul class="editor_modes action_list action_modes editor_tabs">
-        <li class="view-button oa_editor_tab" id="oa_edit_prompt_tab"><a href="#oa_prompt_editor_wrapper">{% trans "Prompt" %}</a></li>
-        <li class="view-button oa_editor_tab" id="oa_edit_rubric_tab"><a href="#oa_rubric_editor_wrapper">{% trans "Rubric" %}</a></li>
-        <li class="view-button oa_editor_tab" id="oa_edit_schedule_tab"><a href="#oa_schedule_editor_wrapper">{% trans "Schedule" %}</a></li>
-        <li class="view-button oa_editor_tab" id="oa_edit_assessment_steps_tab"><a href="#oa_assessment_steps_editor_wrapper">{% trans "Assessment steps" %}</a></li>
-        <li class="view-button oa_editor_tab" id="oa_edit_settings_tab"><a href="#oa_settings_editor_wrapper">{% trans "Settings" %}</a></li>
+        <li role="tab" class="view-button oa_editor_tab" id="oa_edit_prompt_tab">
+            <a href="#oa_prompt_editor_wrapper">
+                <span class="validation-warning" aria-label="">{% include 'openassessmentblock/icons/warning_filled.html' %}</span>
```
Comment (leangseu-edx) [inline openassessment/templates/openassessmentblock/edit/oa_edit_header_and_validation.html:7]: should this include `aria-label="warning"`?

Comment (nsprenkle) [reply openassessment/templates/openassessmentblock/edit/oa_edit_header_and_validation.html:7]: @wittjeff's requirements said to leave that as `aria-label=""`
